packages/next/src/server/lib/router-server.ts

```diff
@@ -679,6 +679,12 @@ export async function initialize(opts: {
         'no-cache, no-store, max-age=0, must-revalidate'
       )
 
+      if (opts.dev && !matchedOutput && parsedUrl.pathname === '/favicon.ico') {
```

Comment:
```suggestion
      // Short-circuit favicon.ico serving so that the 404 page doesn't get built as favicon is requested by the browser when loading any route.
      if (opts.dev && !matchedOutput && parsedUrl.pathname === '/favicon.ico') {
```